platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,296 @@
+"""
+Django models related to the Tasking system
+"""
+import json
+
+from django.db import models
+
+from pulp.platform.models import Model
+
+class ReservedResource(Model):
+    """
+    Resources that have been reserved
+
+    Fields:
+
+    :cvar resource: The name of the resource reserved for the task.
+    :type resource: models.TextField
+
+    Relations:
+
+    :cvar task: The task associated with this reservation
+    :type task: models.ForeignKey
+
+    :cvar worker: The worker associated with this reservation
+    :type worker: models.ForeignKey
+    """
+    resource = models.TextField()
+
+    task = models.ForeignKey(Task, unique=True)
+    worker = models.ForeignKey(Worker)
```

Comment:
Instead of utilizing strings for the task_id and worker_name, I converted these fields into foreign keys to tie them to the task and worker.  This is in line with converting to a relational model, but might need to be revisited if it causes headaches for the reserved resource code.